samples/stats/main.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2017, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/census-instrumentation/opencensus-go/stats"
+	"github.com/census-instrumentation/opencensus-go/tags"
+	"golang.org/x/net/context"
+)
+
+func main() {
+	// ---------------------------------------------
+	// CREATING/REGISTERING KEYS AND VIEWS
+	// ---------------------------------------------
+
+	// Creates keys
+	key1, err := tags.CreateKeyString("/widget.com/key/deviceTypeID")
+	if err != nil {
+		log.Fatalf("Key '/widget.com/key/deviceTypeID' not created %v", err)
+	}
+	key2, err := tags.CreateKeyString("/widget.com/key/osVersionID")
+	if err != nil {
+		log.Fatalf("Key '/widget.com/key/osVersionID' not created %v", err)
+	}
+
+	// Create measures
+	mf, err := stats.NewMeasureFloat64("/widget.com/measure/float64/video_size", "size of video document processed in megabytes (10**6)", "MBy")
+	if err != nil {
+		log.Fatalf("Measure '/widget.com/measure/float64/video_size' not created %v", err)
+	}
+	mi, err := stats.NewMeasureInt64("/widget.com/measure/int64/video_spam_count", "count of videos marked as spam/inappropriate", "1")
+	if err != nil {
+		log.Fatalf("Measure '/widget.com/measure/int64/video_spam_count' not created %v", err)
+	}
+
+	// Create aggregations
+	histogramBounds := []float64{-10, 0, 10, 20}
+	agg1 := stats.NewAggregationDistribution(histogramBounds)
+	agg2 := stats.NewAggregationCount()
+
+	duration := 10 * time.Second
+	precisionIntervals := 10
+	wnd1 := stats.NewWindowSlidingTime(duration, precisionIntervals)
+
+	// Create views
+	myView1 := stats.NewView("/widget.com/view/video_size/distribution", "a distribution of video sizes processed tagged by device and os", []tags.Key{key1, key2}, mf, agg1, wnd1)
```

Comment:
"/mycompany.com/..." would be more obvious as a namespace.

Reply:
Done